### PR TITLE
Add WalletConnect functionality (enable, disable, sendTransaction)

### PR DESCRIPTION
**Pull request by ccali11**
@shanejearley - Please see WalletConnect code here. I haven't figured out yet how to test on a test net because I am using Metamask and Trust Wallet on mobile to connect to Casimir. I could do some more investigation, but wouldn't be surprised if test nets on mobile are harder to get set up. Either way, the enable, disable, and sendTransaction functionalities are communicating appropriately and figured I'd submit a PR to build some momentum again here.

Please let me know if this file structure works too; you'll likely have some good thoughts on how to better structure it instead of the way I did (breaking out WalletConnect methods into a `utils/` directory). Also, there is an outstanding yet simple TS issue that I could use your opinion on how to resolve quickly. I think you will know exactly what to do with it.

### CODE REVIEW

**Comment by ccali11** [comment]
@shanejearley - This PR is now ready for approval. If you are able to merge tomorrow morning your time, I will create a new branch off develop and get rolling with Ledger tomorrow morning my time. 👍

File: apps/web/src/composables/walletConnect.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import WalletConnect from '@walletconnect/client'
+import QRCodeModal from '@walletconnect/qrcode-modal'
+import { ref, Ref } from 'vue'
+import { ethers } from 'ethers'
+
+export default function useWalletConnect() {
+  const connector: Ref<WalletConnect | undefined> = ref()
```
Comment (shanejearley) [inline apps/web/src/composables/walletConnect.ts:7]: Cool, good move.

File: apps/web/src/composables/wallet.ts [hunk 2]
```diff
@@ -3,20 +3,23 @@ import { ethers } from 'ethers'
...
 }
-
 const ethersProviderList = ['MetaMask', 'CoinbaseWallet']
+
+const { requestEthersAccount } = useEthers()
+const { enableWalletConnect, disableWalletConnect, sendWalletConnectTx } =
+  useWalletConnect()
+
+const amount = ref<string>('0.01')
+const toAddress = ref<string>('0xD4e5faa8aD7d499Aa03BDDE2a3116E66bc8F8203')
```
Comment (shanejearley) [inline apps/web/src/composables/wallet.ts:22]: Will update these when we meet based on HH changes.

File: apps/web/src/composables/wallet.ts [hunk 3]
```diff
@@ -65,9 +70,20 @@ export default function useWallet() {
     }
   }
 
+  // TODO: Fold this into the logic of switching to/from other wallet provider depending on front-end implementation
```
Comment (shanejearley) [inline apps/web/src/composables/wallet.ts:73]: Or in the current connectWallet method?

Comment (shanejearley) [reply apps/web/src/composables/wallet.ts:73]: Like:
`if (selectedProvider.value && selectedProvider.value === 'WalletConnect') { ... }`

File: apps/web/src/composables/walletConnect.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      try {
+        connector.value?.killSession()
+      } catch (error) {
+        console.log(`disconnect error in listener :>> ${error}`)
+      }
+    })
+    return
+  }
+
+  async function sendWalletConnectTx(amount: string, toAddress: string) {
```
Comment (shanejearley) [inline apps/web/src/composables/walletConnect.ts:47]: Let's consistently use either `Transaction` or `Tx`. Looks like ethers uses the former, and we have `sendIoPayTransaction`, so maybe let's go with that.

**Review by shanejearley — CHANGES_REQUESTED** [review]

**Comment by shanejearley** [comment]
Looks great. I requested a few small changes and then added a few other changes to support the local Hardhat networks on Metamask mobile. Haven't had a chance to write up usage documentation yet, but generally:
- `npm run dev:ethereum` in another terminal to run an Ethereum localnet (more info in our README.md [here](https://github.com/consensusnetworks/casimir#local-nodes))
- You'll see values after "Your mnemonic is" and "Your local tunnel is" printed out during the startup followed by a list of account addresses
- Grab the mnemonic and import it into Metamask (web or mobile) – the initial account will match index 0 of the printed accounts and each newly created account will match the others (derived from the same seed)
- On web, add the printed RPC URL (defaults to `http://127.0.0.1:8545`) to a custom network [arbitrarily] named `Hardhat` with chain ID 1337 and symbol ETH
- On mobile, add the printed local tunnel URL (should be `https://cn-hh-${your-computer-hostname}.loca.lt`) with the same chain ID and symbol as above
- You should now have 32 ETH on each account

**Comment by shanejearley** [comment]
^Tunnelled RPC URL switched to `https://cn-hh-${your-computer-**username**}.loca.lt`.

**Review by shanejearley — APPROVED** [review]

**Pull request merged**